core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPreferWritePartitioning.java

```diff
@@ -93,6 +104,52 @@ public void testDoesNotFire()
                 .doesNotFire();
     }
 
+    @Test
+    public void testThreshold()
```

Comment:
What does it test? Why it did not fire?

Reply:
It fires without a threshold, so when the threshold `2` is added it should not fire since there is only one partition

Reply:
please add a comment